auction-server/src/watcher.rs

```diff
@@ -29,26 +52,75 @@ pub async fn run_watcher_loop_svm(store: Arc<Store>, chain_id: String) -> Result
         .ok_or(anyhow!("Chain not found: {}", chain_id))?;
 
     while !SHOULD_EXIT.load(Ordering::Acquire) {
-        let response = chain_store
-            .client
-            .get_latest_blockhash_with_commitment(CommitmentConfig::finalized())
-            .await;
-
-        match response {
-            Ok(result) => store.broadcast_svm_chain_update(SvmChainUpdate {
-                chain_id:  chain_id.clone(),
-                blockhash: result.0,
-            }),
-            Err(e) => {
+        match (
+            chain_store
+                .client
+                .get_latest_blockhash_with_commitment(CommitmentConfig::finalized())
+                .await,
+            get_median_prioritization_fee(
+                &chain_store.client,
+                chain_store.config.prioritization_fee_percentile,
+            )
+            .await,
+        ) {
+            (Ok(result), Ok(median_fee)) => {
+                store.broadcast_svm_chain_update(SvmChainUpdate {
+                    chain_id:  chain_id.clone(),
+                    blockhash: result.0,
+                });
+                chain_store.add_recent_prioritization_fee(median_fee).await;
+            }
+            (Err(e), _) => {
                 return Err(anyhow!(
                     "Polling blockhash failed for chain {} with error: {}",
                     chain_id,
                     e
                 ));
             }
+            (_, Err(e)) => {
+                return Err(anyhow!(
+                    "Polling prioritization fees failed for chain {} with error: {}",
+                    chain_id,
+                    e
+                ));
+            }
         }
 
         tokio::time::sleep(GET_LATEST_BLOCKHASH_INTERVAL).await;
     }
     Ok(())
 }
+
+pub async fn get_median_prioritization_fee(
+    client: &RpcClient,
+    percentile: Option<u64>,
+) -> Result<u64, ClientError> {
+    let accounts: Vec<String> = vec![];
+    let mut args: Vec<serde_json::Value> = vec![serde_json::to_value(accounts)?];
+
+    if let Some(percentile) = percentile {
+        args.push(serde_json::json!({ "percentile": percentile }));
+    }
+
+    fn median(values: &mut [u64]) -> u64 {
+        let mid = values.len() / 2;
+        *values.select_nth_unstable(mid).1
+    }
+
+    client
+        .send(
+            solana_client::rpc_request::RpcRequest::GetRecentPrioritizationFees,
+            serde_json::Value::from(args),
+        )
+        .await
+        .map(|mut values: Vec<RpcPrioritizationFee>| {
+            values.sort_by(|a, b| b.slot.cmp(&a.slot));
```

Comment:
can we use sth like `sort_by_key(|a| a.slot)`?

Reply:
I tried this. Doesn't work because I want the array to be sorted in decreasing order with respect to slot to be able to use `take()` later.
You suggestion would sort the vec in increasing order.